platform/linux-generic/odp_ipsec_sad.c

```diff
@@ -906,41 +935,91 @@ int _odp_ipsec_sa_replay_precheck(ipsec_sa_t *ipsec_sa, uint32_t seq,
 int _odp_ipsec_sa_replay_update(ipsec_sa_t *ipsec_sa, uint32_t seq,
 				odp_ipsec_op_status_t *status)
 {
-	int cas = 0;
-	uint64_t state, new_state;
+	uint32_t bucket, wintop_bucket, new_bucket;
+	uint32_t bkt_diff, bkt_cnt;
+	uint64_t bit = 0;
 
 	if (!ipsec_sa->antireplay)
 		return 0;
 
-	state = odp_atomic_load_u64(&ipsec_sa->hot.in.antireplay);
+	if ((seq + ipsec_sa->hot.in.win_size) <=
+	    (odp_atomic_load_u64(&ipsec_sa->hot.in.wintop_seq) & 0xffffffff)) {
+		status->error.antireplay = 1;
+		return -1;
+	}
```

Comment:
The checking here does not fix the race condition that I commented about. A packet may pass this test but fall below the window before the actual window update part later in this function. Also, this check is done also for the 32-packet window code path even though it is not needed in that case (because the check is done in the existing code).

Reply:
Correct, this check is not required for WS 32 case, will correct it.